### PR TITLE
Merging master to custom curve mesh

### PR DESCRIPTION
**Pull request by jonathanraes**

### CODE REVIEW

File: src/main/java/com/funkydonkies/controllers/PhysicsController.java [hunk 1]
```diff
@@ -14,11 +14,13 @@
  */
 public class PhysicsController extends RigidBodyControl implements
```
Comment (Danilodum14) [inline src/main/java/com/funkydonkies/controllers/PhysicsController.java]: Missing javadoc

File: src/main/java/com/funkydonkies/obstacles/Target.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	
+	/**
+	 * Respawn the target at a reachable location.
+	 * TODO make the spawn location random and make sure its reachable
+	 */
+	public void respawn() {
+		final float x = (float) Math.random() * (SplineCurveController.POINT_DISTANCE * 
+				SplineCurveController.getAmountOfControlPoints());
+		
+		final float y = (float) Math.random() * SplineCurveController.POINTS_HEIGHT + 20;
```
Comment (Danilodum14) [inline src/main/java/com/funkydonkies/obstacles/Target.java]: hardcoding needs var

File: src/main/java/com/funkydonkies/obstacles/ObstacleFactory.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import com.jme3.asset.AssetManager;
+import com.jme3.scene.Node;
+
+/**
+ * ObstacleFactory is the class for obstacles.
+ * @author SDumasy
+ *
+ */
+public class ObstacleFactory {
+	public static final float obstacleWidth = 20;
```
Comment (Danilodum14) [inline src/main/java/com/funkydonkies/obstacles/ObstacleFactory.java]: bad names

File: src/main/java/com/funkydonkies/w4v3/App.java [hunk 4]
```diff
@@ -69,32 +72,30 @@ public void simpleInitApp() {
...
-		final int obstacleDepth = 1;
-		
-//		clBox = (MovingBox) facto.makeObstacle("MOVINGBOX", obstacleWidth, obstacleHeight, 
-//				obstacleDepth);
-		
 		final Material mat2 = new Material(assetManager, UNSHADED_MATERIAL_PATH);
 		mat2.setColor(COLOR, ColorRGBA.Red);
-//		clBox.draw(mat2, getPhysicsSpace(), rootNode);
-		
+		movBox.draw(mat2, getPhysicsSpace());
```
Comment (Danilodum) [inline src/main/java/com/funkydonkies/w4v3/App.java:90]: movBox will be moved out of App after architecture refactor

**Comment by olivier-dikken** [comment]
balls don't collide correctly with obstacle. wave height does not correspond to detected height. Sometimes target spawns out of bounds. Otherwise it works as expected.

**Pull request merged**